File: src/lecture2/lecture2_dr_cpp/src/dr_node.cpp
#include <ros/ros.h>
#include <dynamic_reconfigure/server.h>
#include <lecture2_dr_cpp/DRConfig.h>

void callback(lecture2_dr_cpp::DRConfig &config, uint32_t level)
{
    ROS_INFO_STREAM(config.int_param << " " << config.double_param << " " << config.str_param << " " << (config.bool_param ? "True" : "False") << " " << config.enum_param);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "dr_node");
    dynamic_reconfigure::Server<lecture2_dr_cpp::DRConfig> server;
    server.setCallback(boost::bind(&callback, _1, _2));
    ros::spin();
    return 0;
}

File: src/lecture2/lecture2_dr_py/scripts/dr_node.py
#!/usr/bin/env python3
import rospy
from dynamic_reconfigure.server import Server
from lecture2_dr_py.cfg import DRConfig

def callback(config, level):
    rospy.loginfo("""{int_param} {double_param} {str_param} {bool_param} {enum_param}""".format(**config))
    return config

if __name__ == "__main__":
    rospy.init_node("dr_node")
    srv = Server(DRConfig, callback)
    rospy.spin()

